sympy/sets/sets.py

```diff
@@ -218,16 +218,12 @@ def _complement(self, other):
 
         elif isinstance(other, FiniteSet):
             from sympy.utilities.iterables import sift
+            from sympy.core.logic import fuzzy_bool
 
-            def ternary_sift(el):
-                contains = self.contains(el)
-                return contains if contains in [True, False] else None
-
-            sifted = sift(other, ternary_sift)
+            sifted = sift(other, lambda x: fuzzy_bool(x))
```

Comment:
You can directly pass `fuzzy_bool` function like `sifted = sift(other, fuzzy_bool)`

Reply:
This is probably not the way `fuzzy_bool` should be used. It seems to recognize 0 and 1 but no other numbers:
 ```
In [10]: fuzzy_bool(0)
Out[10]: False

In [11]: fuzzy_bool(1)
Out[11]: True

In [12]: fuzzy_bool(2)

In [13]: fuzzy_bool(3)

```